### ISSUE TITLE
Type inference failed: infix fun <T> T.shouldBe(matcher: Matcher<T>): Unit cannot be applied to receiver: Int  arguments: (Int)

### ISSUE THREAD

**Issue by elect86** [kotlintest/kotlintest#301]
Example code

```
            val a = 4
            val b = 5
            a shouldBe b
```

This happens with the last version of Kotlintest

2.0.7 was fine

Edit: it appears that 

`import io.kotlintest.shouldBe` 

works, but the automatically added by Idea

`import io.kotlintest.matchers.shouldBe`

doesn't

**Comment by sksamuel** [comment]
3.0.4 ?

**Comment by elect86** [comment]
Yep

**Comment by sksamuel** [comment]
I've released 3.0.5 which has this fixed.

**Comment by elect86** [comment]
excellent, thanks!

**Comment by sschuberth** [comment]
What's the difference between `io.kotlintest.shouldBe` and `io.kotlintest.matchers.shouldBe` anyway? I've found only the former to be working with kotlintest 3.x in my code, but I didn't find any docs / migration guide about it.

**Comment by sksamuel** [comment]
They both should work (but one is deprecated), but it was broken until 3.0.6.

**Comment by elect86** [comment]
A small postscript

`io.kotlintest.matchers.shouldBe` still triggers

> Cannot choose among the following candidates without completing type inference: public infix fun  Int.shouldBe(any: Any?): Unit defined in io.kotlintest.matchers public infix fun  Int.shouldBe(any: Int?): Unit defined in io.kotlintest.matchers

**Comment by sschuberth** [comment]
@elect86 I believe that's exactly the bogus error message (which I was also getting) that has been fixed with version 3.0.6.

**Comment by sksamuel** [comment]
Are you using 3.0.6 ?

**Comment by sksamuel** [comment]
Snap ^^

**Comment by elect86** [comment]
No, only the `matchers` one (and I switched to the other one to fix it)

**Comment by sksamuel** [comment]
Ok I'll take another look but I was convinced it was fixed in 3.0.6 :) Was it only with ints you had an issue?

**Comment by elect86** [comment]
I had 3/4 consecutive warning, all ints in that class

Btw, I switched to the other import, it's fine for me now, I just wanted to signal the thing in case something popups again in the future :)
